samples/nrf9160/modem_shell/src/link/link.c

```diff
@@ -82,24 +82,30 @@ static int32_t modem_rsrp = LINK_RSRP_VALUE_NOT_KNOWN;
 struct ncellmeas_data {
 	struct k_work_delayable work;
 
+	struct lte_lc_ncellmeas_params params;
+
 	enum link_ncellmeas_modes mode;
-	enum lte_lc_neighbor_search_type search_type;
 	int periodic_interval;
 };
 static struct ncellmeas_data ncellmeas_work_data;
-
-static enum link_ncellmeas_modes ncellmeas_mode = LINK_NCELLMEAS_MODE_NONE;
-static enum lte_lc_neighbor_search_type ncellmeas_search_type = LTE_LC_NEIGHBOR_SEARCH_TYPE_DEFAULT;
-static int ncellmeas_periodic_interval;
+static struct ncellmeas_data ncellmeas_param_data = {
+	.mode = LINK_NCELLMEAS_MODE_NONE,
+	.params.search_type = LTE_LC_NEIGHBOR_SEARCH_TYPE_DEFAULT,
+	.periodic_interval = 0,
+};
```

Comment:
I was trying to see that do we need `ncellmeas_param_data`. I cannot make a conclusion on this so I'll just leave an open comment and you can check that as you know the details.

I will add few more comments related to `ncellmeas_work_data` and `ncellmeas_param_data` where I can see that `ncellmeas_param_data` might not be needed.

Reply:
I think that is good to have as a container for all related params, both mosh + lte_lc

Reply:
I understand we need a container so `struct ncellmeas_data` is well justified. But what I couldn't figure out from the code is why we need both `ncellmeas_work_data` and `ncellmeas_param_data`? And we seem to copy most of `ncellmeas_param_data` variables into `ncellmeas_work_data` when the data is actually scheduled for the work.